File: server/lms-atom/src/main/java/com/lms/atom/language/service/LanguageService.java
package com.lms.atom.language.service;

import com.lms.atom.exception.AtomException;
import com.lms.common.dto.atom.language.LanguageDTO;
import com.lms.common.dto.response.ListResult;

public interface LanguageService {

    ListResult<LanguageDTO> find(String query, int limit, int offset) throws Exception;

    ListResult<LanguageDTO> find(Long id, String code, String name, int limit, int offset) throws Exception;

    LanguageDTO update(LanguageDTO language) throws AtomException;

    LanguageDTO save(LanguageDTO language) throws AtomException;

    void delete(Long id) throws Exception;
}

File: admin-panel/src/app/services/language.service.ts
import {Injectable} from '@angular/core';

import 'rxjs/add/operator/toPromise';

import {BaseService} from "./base.service";
import {ActionResponse} from "../model/response/action-response";
import {ActionResponseWithData} from "../model/response/action-response-with-data";
import {ListResult} from "../model/response/list-result";
import {Language} from "../model/resources/language";

@Injectable()
export class LanguageService {

  private url = 'atom/language-api/';

  constructor(private baseService: BaseService<Language>) {
  }

  findLanguages(request): Promise<ActionResponseWithData<ListResult<Language>>> {
    return this.baseService.getAll(this.url + "find", request);
  }

  findLanguagesQuick(request): Promise<ActionResponseWithData<ListResult<Language>>> {
    return this.baseService.getAllByParams(this.url + "quick-find", request);
  }

  create(language: Language): Promise<ActionResponseWithData<Language>> {
    return this.baseService.create(this.url + "save", language);
  }

  update(language: Language): Promise<ActionResponseWithData<Language>> {
    return this.baseService.update(this.url + "update", language);
  }

  delete(id: number): Promise<ActionResponse> {
    return this.baseService.delete(this.url + "language", id);
  }

  private handleError(error: any): Promise<any> {
    console.error('Error', error);
    return Promise.reject(error.message || error);
  }
}
